art-blocks-101/post-mint-parameters.md

```diff
@@ -0,0 +1,192 @@
+
+- Parameter names
+- Parameter values
+- Parameter types
+- Parameter constraints/limits
+- Parameter lock dates
+
+Note that if you would like your parameters to show up as feature overrides, using the same parameter name as a feature name will properly override the feature.
+
+## Configuring token as collector
+
+Token owners may log into the artblocks.io website and configure parameters for their token on the token's page. The website will display a preview of the token with edited values, allowing collectors to preview the changes before forever registering the changes on the blockchain as part of the token's provenance. The connected wallet must have authority to change parameters for the token, or be a valid delegate for the token on delegate.xyz v2.
+
+# Technical Overview
+
+## Compatibility
+
+Recent Art Blocks Engine Flex NFTs (core version v3.2.5+) support configurable parameters. If you would like an updated core contract, please contact the Art Blocks team.
+
+## Supported parameter types
+
+- Enum
+- Boolean
+- Integer Number Range
+- Decimal Number Range
+- Color
+- Timestamp (1970+)
+- String
+
+## Supported Authorization Methods
+
+- Artist
+- Token Owner
+- Address/Smart Contract
+
+## Supported Parameter Constraints
+
+- Min
+- Max
+- Lock date
+
+## Accessing parameters in project script
+
+Configurable parameters are injected using the typical Art Blocks Generator's `tokenData` object in a Flex Dependency's data field.
+
+The following is a simple example of how to access an integer parameter in your project script:
+
+```javascript
+// Typically, the first external asset dependency is the PMP.
+const pmp = tokenData.externalAssetDependencies[0];
+
+// helper function to parse PMP integer values, or fallback
+// to the token-hash-based value if the PMP value is not set
+function getPmpInt(pmpVal, hashBasedVal) {
+  const n = parseInt(pmpVal);
+  return isNaN(n) ? hashBasedVal : n;
+}
+
+// accessing a PMP named "NumShapes"
+const amountMoving = getPmpInt(
+  pmp?.data?.["NumShapes"], // prioritize the PMP value if defined
+  R.random_int(0, 100) // R is the hash-based pseudorandom number generator
+);
+```
+
+The following is a more complete example of how to access multiple parameters of different types in your project script.
+
+```javascript
+// Palettes are defined in the script, as well as in the PMP Configuration as an enum
+let palettes = [
+  {
+    name: "Midnight Citrus",
+    colors: ["#0f0f1c", "#ffcc00", "#ff7f11", "#ffa69e", "#4d194d"],
+  },
+  {
+    name: "Electric Bloom",
+    colors: ["#8ecae6", "#219ebc", "#023047", "#ffb703", "#fb8500"],
+  },
+  {
+    name: "Forest Neon",
+    colors: ["#073b3a", "#0b6e4f", "#08a045", "#d4d700", "#fdfcdc"],
+  },
+];
+
+// Typically, the first external asset dependency is the PMP.
```

Comment:
Might be worth adding something about how to definitively determine which external asset dependency is the PMP in case it's not the first

Reply:
updated in [61c7c14](https://github.com/ArtBlocks/artblocks-docs/pull/328/commits/61c7c148350287973f20047a24a20d7cc2cbe5de)